python/cudf/cudf/core/column/column.py

```diff
@@ -1503,17 +1503,11 @@ def column_empty(
             ),
         )
     elif dtype.kind in "OU" and not isinstance(dtype, DecimalDtype):
-        data = None
+        data = as_buffer(
+            rmm.DeviceBuffer(size=row_count * cudf.dtype("int8").itemsize)
+        )
         children = (
             full(row_count + 1, 0, dtype=libcudf.types.size_type_dtype),
-            build_column(
-                data=as_buffer(
-                    rmm.DeviceBuffer(
-                        size=row_count * cudf.dtype("int8").itemsize
```

Comment:
Question: does `dtype.kind in "OU" and not isinstance(dtype, DecimalDtype):` allow only string type?
I didn't change the logic here. For an empty string column, why do we need a chars buffer of size `row_count` ? it could be an empty buffer since the offsets are all `0`.

Reply:
In general this condition can also be true for (at least) List and Struct columns. But, those are handled by specific cases above.

From a quick test, I think we don't need an empty chars buffer, so can you try using `rmm.DeviceBuffer(0)`?